tests/core/test_playback.py

```diff
@@ -789,3 +808,65 @@ def test(self):
         c.playback.pause()
         c.playback.next()
         b.playback.change_track.assert_called_once_with(track2)
+
+
+class Bug1358RegressionTest(unittest.TestCase):
+
+    def setUp(self):  # noqa: N802
+        config = {
+            'core': {
+                'max_tracklist_length': 10000,
+            }
+        }
+
+        self.backend1 = mock.Mock()
+        self.backend1.uri_schemes.get.return_value = ['dummy1']
+        self.playback1 = mock.Mock(spec=backend.PlaybackProvider)
+        self.backend1.playback.change_track.return_value.get.return_value = \
+            False
+        self.backend1.playback = self.playback1
+
+        self.backend2 = mock.Mock()
+        self.backend2.uri_schemes.get.return_value = ['dummy2']
+        self.playback2 = mock.Mock(spec=backend.PlaybackProvider)
+        self.backend1.playback.change_track.return_value.get.return_value = \
+            False
+        self.backend2.playback = self.playback2
+
+        self.tracks = [
+            Track(uri='dummy1:a', length=40000),
+            Track(uri='dummy2:a', length=40000),
+        ]
+
+        self.uris = [
+            'dummy1:a', 'dummy2:a']
+
+        self.core = core.Core(config, mixer=None,
+                              backends=[self.backend1, self.backend2])
+
+        def lookup(uris):
+            result = {uri: [] for uri in uris}
+            for track in self.tracks:
+                if track.uri in result:
+                    result[track.uri].append(track)
+            return result
+
+        self.lookup_patcher = mock.patch.object(self.core.library, 'lookup')
+        self.lookup_mock = self.lookup_patcher.start()
+        self.lookup_mock.side_effect = lookup
+
+        self.core.tracklist.add(uris=self.uris)
+
+        self.tl_tracks = self.core.tracklist.tl_tracks
+
+    def tearDown(self):  # noqa: N802
+        self.lookup_patcher.stop()
+
+    def test_next_in_consume_mode_removes_unplayable_track(self):
+        self.core.tracklist.consume = True
+
+        self.core.playback.play(self.tl_tracks[0])
+        self.core.playback.next()
+
+        self.assertNotIn(self.tl_tracks[0], self.core.tracklist.tl_tracks)
```

Comment:
Similarly, this should use `get_tl_tracks()` as the `tl_tracks` property is deprecated.